### PR TITLE
Week 2 Finished

### PR DESCRIPTION
**Pull request by ConnorNovak**
Week 2 code is finished and committed, no bells and-or whistles.

### CODE REVIEW

File: gene_finder.py [hunk 1]
```diff
@@ -2,7 +2,8 @@
 """
 YOUR HEADER COMMENT HERE
```
Comment (segerphilip) [inline gene_finder.py]: Either include some helpful info here (if something does not work, what tripped you up if anything) or remove.

File: gene_finder.py [hunk 2]
```diff
@@ -12,155 +13,368 @@
 
 
 def shuffle_string(s):
-    """Shuffles the characters in the input string
-        NOTE: this is a helper function, you do not
-        have to modify this in any way """
+    """
```
Comment (segerphilip) [inline gene_finder.py]: Nice docstrings 👍

File: gene_finder.py [hunk 3]
```diff
@@ -12,155 +13,368 @@
...
+        """
+    if (nucleotide == "A") or (nucleotide == "a"):
+        return "T"
+    elif (nucleotide == "T") or (nucleotide == "t"):
+        return "A"
+    elif (nucleotide == "C") or (nucleotide == "c"):
+        return "G"
+    elif (nucleotide == "G") or (nucleotide == "g"):
+        return "C"
+    else:
```
Comment (segerphilip) [inline gene_finder.py]: Also good edge case checking

File: gene_finder.py [hunk 4]
```diff
@@ -12,155 +13,368 @@
...
+        Two examples to test nominal functionality and two error test, one for
+        and int input, one for a string input containing a nonsense character
+        """
+    # Argument error checks
+    if (type(dna) != str):
+        print("ERROR: expected string, recieved other type")
+        return
+
+    # Flips dna string and gets complements
+    reverse = dna[::-1]
```
Comment (segerphilip) [inline gene_finder.py]: 👏

File: gene_finder.py [hunk 5]
```diff
@@ -12,155 +13,368 @@
...
+    aa_list = []
+    for str in orf_list:
+        if (len(str) > threshold):
+            str = coding_strand_to_AA(str)
+            aa_list.append(str)
+
+    print('Found:',len(aa_list),'amino acid lists')
+
+    # Writes amino acids to text file aminoacids.txt
+    f = open('aminoacids.txt','w')
```
Comment (segerphilip) [inline gene_finder.py]: Nice, works well
